### PR TITLE
jackgopack4/OTEL-1955 infraattributes processor

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?

Removes dependency on tagger and taggerimpl packages for infraattributes processor by using the interface
### Motivation

OTEL-1955
### Additional Notes

unit tests passed under otel folder and in CI
### Possible Drawbacks / Trade-offs

none, no major changes
### Describe how to test/QA your changes

run unit tests in go/python

### CODE REVIEW

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package infraattributesprocessor
+
+import "github.com/DataDog/datadog-agent/comp/core/tagger/types"
+
+type taggerClient interface {
+	Tag(entity string, cardinality types.TagCardinality) ([]string, error)
+	GlobalTags(cardinality types.TagCardinality) ([]string, error)
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient.go:12]: Can you add comments for each of these functions to describe their usage? i.e.
```
// Tag does ...
Tag(entity string)...
```

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"github.com/DataDog/datadog-agent/comp/core/tagger/taggerimpl/collectors"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+)
+
+type testTaggerClient struct {
+	m map[string][]string
+}
+
+func (t *testTaggerClient) Tag(entityID string, _ types.TagCardinality) ([]string, error) {
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go:17]: Same here, can you add comments for these functions?

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+package infraattributesprocessor
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/tagger/taggerimpl/collectors"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+)
+
+type testTaggerClient struct {
+	m map[string][]string
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go:14]: nit: a more descriptive name than `m` would be helpful here

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics_test.go [hunk 4]
```diff
@@ -128,12 +127,13 @@ func TestInfraAttributesMetricProcessor(t *testing.T) {
...
 			}
-			fakeTagger := taggerimpl.SetupFakeTagger(t)
-			defer fakeTagger.ResetTagger()
-			fakeTagger.SetTags("container_id://test", "test", []string{"container:id"}, nil, nil, nil)
-			fakeTagger.SetTags("deployment://namespace/deployment", "test", []string{"deployment:name"}, nil, nil, nil)
-			fakeTagger.SetTags(collectors.GlobalEntityID, "test", []string{"global:tag"}, nil, nil, nil)
-			factory := NewFactory(fakeTagger)
+			tc := &testTaggerClient{
+				m: make(map[string][]string),
+			}
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics_test.go:132]: Could you add a constructor in `taggerclient_test.go` to simplify instantiating this struct throughout the test files?

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient.go (hunk 1 above)
Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient.go]: good to go thanks

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go (hunk 2 above)
Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go:26]: thanks, have done it

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go (hunk 3 above)
Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go]: great feedback, made a change.

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics_test.go (hunk 4 above)
Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics_test.go]: done, great idea thanks!

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+// newTestTaggerClient creates and returns a new testTaggerClient with an empty string map
+func newTestTaggerClient() *testTaggerClient {
+	return &testTaggerClient{
+		tagMap: make(map[string][]string),
+	}
+}
+
+// Mocks Tag functionality for the purpose of testing, removing dependency on Taggerimpl
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go:25]:
```suggestion
// Tag mocks taggerimpl.Tag functionality for the purpose of testing, removing dependency on Taggerimpl
```

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+func (t *testTaggerClient) Tag(entityID string, _ types.TagCardinality) ([]string, error) {
+	return t.tagMap[entityID], nil
+}
+
+// Mocks GlobalTags functionality for purpose of testing, removing dependency on Taggerimpl
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/taggerclient_test.go:30]:
```suggestion
// GlobalTags mocks taggerimpl.GlobalTags functionality for purpose of testing, removing dependency on Taggerimpl
```

**Review by liustanley — APPROVED** [review]
LGTM, I believe Go convention requires the function name to be the first word in the function comment

**Comment by jackgopack4** [comment]
Ok I'll make that adjustment

**Pull request merged**